frontend/test/metabase/scenarios/collections/permissions.cy.spec.js

```diff
@@ -440,17 +440,40 @@ describe("collection permissions", () => {
         cy.signInAsAdmin();
       });
 
-      it.skip("shouldn't record history steps when there was no diff (metabase#1926)", () => {
+      it("shouldn't render revision history steps when there was no diff (metabase#1926)", () => {
         cy.signInAsAdmin();
         cy.createDashboard("foo").then(({ body }) => {
           visitAndEditDashboard(body.id);
         });
+
         // Save the dashboard without any changes made to it (TODO: we should probably disable "Save" button in the first place)
         saveDashboard();
-        // Take a look at the generated history - there shouldn't be anything other than "First revision" (dashboard created)
-        cy.icon("ellipsis").click();
-        cy.findByText("Revision history").click();
-        cy.findAllByRole("button", { name: "Revert" }).should("not.exist");
+        cy.icon("pencil").click();
+        saveDashboard();
+
+        openRevisionHistory();
+
+        const revisionHistoryEntries = cy
+          .findAllByText("Bobby Tables")
+          .parent();
+
+        // This accounts for:
+        // 1. First revision.
+        // 2. added a card.
+        // 3. rearranged the cards. (TODO: consider fixing this as cards we never rearranged)
+
+        // Revision history entries from saving no change
+        // should not be rendered.
+        revisionHistoryEntries.should("have.length", 3);
+
+        // Topmost visible revision history entry
+        // should not have a Revert button, as it's
+        // the latest practical one.
+        revisionHistoryEntries.first().within(() => {
+          cy.findByText("Revert").should("not.exist");
+        });
+
+        cy.findAllByText("Revert").should("have.length", 2);
```

Comment:
Let's please stick to using selectors from [Cypress testing library](https://testing-library.com/docs/cypress-testing-library/intro/) because they enforce good a11y practices.
`cy.findAllByRole("button", { name: "Revert" }).should(...)`